app/code/community/Payone/Core/Helper/Ratepay.php

```diff
@@ -0,0 +1,335 @@
+<?php
+/**
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the GNU General Public License (GPL 3)
+ * that is bundled with this package in the file LICENSE.txt
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Payone_Core to newer
+ * versions in the future. If you wish to customize Payone_Core for your
+ * needs please refer to http://www.payone.de for more information.
+ *
+ * @category        Payone
+ * @package         Payone_Core_Helper
+ * @subpackage
+ * @copyright       Copyright (c) 2020 <kontakt@fatchip.de> - www.fatchip.de
+ * @author          FATCHIP GmbH <kontakt@fatchip.de>
+ * @license         <http://www.gnu.org/licenses/> GNU General Public License (GPL 3)
+ * @link            http://www.fatchip.de
+ */
+
+/**
+ *
+ * @category        Payone
+ * @package         Payone_Core_Helper
+ * @subpackage
+ * @copyright       Copyright (c) 2020 <kontakt@fatchip.de> - www.fatchip.de
+ * @license         <http://www.gnu.org/licenses/> GNU General Public License (GPL 3)
+ * @link            http://www.fatchip.de
+ */
+class Payone_Core_Helper_Ratepay extends Payone_Core_Helper_Abstract
+{
+    const VALIDATION_STEP_POSTALCODE = 'zip';
+    const VALIDATION_STEP_CURRENCY = 'currency';
+    const VALIDATION_STEP_CUSTOMER_AGE = 'customerAge';
+    const VALIDATION_STEP_PHONE_NUMBER = 'phone';
+    const VALIDATION_STEP_BASKET_SIZE = 'basketSize';
+    const VALIDATION_STEP_SHIPPING_ADDRESS = 'shippingAddress';
+
+    const MINIMUM_CUSTOMER_AGE = 18;
+
+    protected $configCache = null;
+
+    /**
+     * @param string $step
+     * @param array $methodsList
+     * @return array
+     */
+    public function filterByValidation($step, $methodsList)
+    {
+        $methodName = 'validate' . ucfirst($step);
+        if (!method_exists($this, $methodName)) {
+            return $methodsList; //FIXME Decide if error when trying to check rule that doesn't exist
```

Comment:
I forgot to mention this question. Do we have an idea about the answer ?

note : the feature is implemented as a modular filter, where we can include steps to checks various point.
The ratepay allowed method list is passed from one step to an other and filtered there.
So far, I decided that if we try to validate a step that doesn't exist (no matching validation function), we skip it by returning the ratepay methods list given as parameter.
It works as a safety measure, when extending the filter with new steps